crates/biome_tailwind_parser/src/lexer/base_name_store.rs

```diff
@@ -0,0 +1,205 @@
+    pub(crate) fn matcher<'s, 't>(&'s self, text: &'t [u8]) -> BaseNameMatcher<'s, 't> {
+        BaseNameMatcher { store: self, text }
+    }
+
+    /// Convenience one-shot API that returns the final base end, including dashed basename + boundary rules.
+    #[inline]
+    pub(crate) fn match_base_end(&self, text: &[u8]) -> usize {
+        self.matcher(text).base_end()
+    }
+}
+
+/// A streaming helper that scans the provided text slice from the beginning
+/// and returns the longest dashed basename prefix end if any.
+///
+/// It stops scanning when:
+/// - It encounters a byte that isn't a path in the trie,
+/// - Or it hits a delimiter (`whitespace`, `!`, `:`).
+///
+/// The returned `usize` is the number of bytes consumed for the matched basename.
+/// The caller is responsible for verifying the boundary (e.g. next byte is `-`, `:`, whitespace, or end-of-input).
+pub(crate) struct BaseNameMatcher<'s, 't> {
+    store: &'s BaseNameStore,
+    text: &'t [u8],
+}
+
+impl<'s, 't> BaseNameMatcher<'s, 't> {
+    pub(crate) fn base_end(&self) -> usize {
+        let mut node_idx = 0usize;
+        let mut best_end: Option<usize> = None;
+
+        // Scan for dashed basename using the trie until a delimiter or mismatch
+        let mut i = 0usize;
+        while i < self.text.len() {
+            let b = self.text[i];
+            if is_delimiter(b) {
+                break;
+            }
+
+            match self.store.find_child(node_idx, b) {
+                Some(next) => {
+                    node_idx = next;
+                    i += 1;
+                    if self.store.nodes[node_idx].terminal {
+                        best_end = Some(i);
+                    }
+                }
+                None => {
+                    break;
+                }
+            }
+        }
+
+        // If we found a dashed match, accept it only if the next byte is a valid boundary or end-of-input
+        if let Some(end) = best_end
+            && self.text.get(end).is_none_or(|b| is_boundary_byte(*b))
+        {
+            return end;
+        }
+
+        // Fallback: naive basename ends at the first '-', whitespace, '!' or ':'
+        let mut j = 0usize;
+        while j < self.text.len() {
+            let b = self.text[j];
+            if b == b'-' || is_delimiter(b) {
+                break;
+            }
+            j += 1;
+        }
+        j
+    }
+}
+
+#[inline]
+fn is_delimiter(b: u8) -> bool {
+    // Delimiters that cannot be part of a basename (excluding '-' which may be inside dashed basenames):
+    // - whitespace
+    // - '!' important modifier
+    // - ':' variant separator
+    matches!(b, b' ' | b'\n' | b'\r' | b'\t' | b'!' | b':')
+}
+
+#[inline]
+fn is_boundary_byte(b: u8) -> bool {
+    // Valid boundary after a dashed basename:
+    // - '-' indicates a value follows
+    // - ':' indicates a variant boundary
+    // - whitespace
+    b == b'-' || b == b':' || matches!(b, b' ' | b'\n' | b'\r' | b'\t')
+}
```

Comment:
Can these functions be turned into `const`? Maybe the compiler can optimise them even further